client/pages/Event/EventDetailsPage.jsx

```diff
@@ -71,23 +83,44 @@ class EventDetailsPage extends React.Component {
       description,
       featuredImage,
     };
+    let message;
     const creator = id === googleId;
+    const isPastEvent = moment.tz(moment.tz.guess()).isAfter(moment.tz(endDate, moment.tz.guess()));
```

Comment:
Great job on this implementation @Alameen688. Works as required. I was however thinking we could improve the readability of this portion of the code and may not even need a third-party package since we are dealing with local timezones. Something like:
```
const localDateTime = new Date(); 
const eventStartDate = new Date(startDate); // returns date object in local tz
const eventEndDate = new Date(endDate);

const isPastEvent = localDateTime > eventEndDate;
const hasCommenced = (eventEndDate > eventStartDate) && (eventStartDate < localDateTime);
...
```
What do you think?

Reply:
Nice suggestion @olusoladavid. However, I would prefer we use a library that is specifically made to handle timezone as there are intricacies when it comes to handling timezone. "Moment timezone contains the data of IANA Time Zone Database to accurately calculate offsets, and provides a variety of APIs that can be used to change and format time zone".

In general, I will look more into how we can handle timezones in the overall application as time goes on but for now I think we should stick with moment-timezone. This is not final, so if you still have any concern based on any finding we could still deliberate on this. Thanks

Reply:
As regards the RSVP button state, I'll look into that now. Thanks, for the feedback

Reply:
Fair point on the use of `moment-timezone` @Alameen688 . I see the library helps us to solve for support in older browsers (IE8) and handling of daylight saving inconsistencies from the linked resource. In that case, we may continue with it. However, I think we can make the current code more readable by assigning the value of `moment.tz.guess()` to a meaningful variable name instead of repeatedly calling it inline. This can also be done for repeated snippets like:
`moment.tz(endDate, moment.tz.guess())` and `moment.tz(startDate, moment.tz.guess())`

Reply:
That's very true. It will be done.

Reply:
Feedback has been implemented @olusoladavid